joint_trajectory_controller/src/joint_trajectory_controller.cpp

```diff
@@ -116,12 +121,33 @@ JointTrajectoryController::state_interface_configuration() const
       conf.names.push_back(joint_name + "/" + interface_type);
     }
   }
+  if (!params_.speed_scaling_state_interface_name.empty())
+  {
+    RCLCPP_INFO(
+      logger, "Using scaling state from the hardware from interface %s.",
+      params_.speed_scaling_state_interface_name.c_str());
+    conf.names.push_back(params_.speed_scaling_state_interface_name);
+  }
   return conf;
 }
 
 controller_interface::return_type JointTrajectoryController::update(
   const rclcpp::Time & time, const rclcpp::Duration & period)
 {
+  if (!params_.speed_scaling_state_interface_name.empty())
+  {
+    if (state_interfaces_.back().get_name() == params_.speed_scaling_state_interface_name)
```

Comment:
You make the assumption that the speed scaling state interface is always the last one in the list. I think this is a rather dangerous assumption

Reply:
Not necessarily, since we [put it there](https://github.com/ros-controls/ros2_controllers/pull/1191/files#diff-8dd2135dc1f4abe6382cc7671ef9ce939196b2d8adf18ac3a77b5c0fddf1a25fR105) But I would prefer a way of resolving this in a cleaner way, yes.